src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -256,10 +255,9 @@ void Connection::activate_session(std::unique_ptr<Session> sess)
 
 void Connection::initiate_session_deactivation(Session* sess)
 {
-    REALM_ASSERT(m_on_idle);
     REALM_ASSERT(&sess->m_conn == this);
     if (REALM_UNLIKELY(--m_num_active_sessions == 0)) {
-        if (m_activated && m_state == ConnectionState::disconnected)
+        if (m_on_idle && m_activated && m_state == ConnectionState::disconnected)
```

Comment:
When would m_on_idle be null?

Reply:
Since it is a `unique_ptr`, I believe I was hitting a crash here during destruction where the trigger had already been destroyed and the event loop was still running. I will remove for now and revisit this when I am wrapping up the "remove stop()" PR.